pmdarima/arima/seasonality.py

```diff
@@ -26,10 +28,124 @@
 
 __all__ = [
     'CHTest',
+    'decompose',
     'OCSBTest'
 ]
 
 
+def decompose(x, type, f, filter=None):
+    """
+
+    Parameters
+    ----------
+    x : np.array
+        The time series of which the trend, seasonal, and noise/random
+        components will be extracted.
+
+    type: str
+        The type of decomposition that will be performed - 'multiplicative' or
+        'additive'. We would use 'multiplicative' generally when we see an
+        increasing trend. We use 'additive' when the trend is relatively
+        stable over time.
+
+    f: int
+        The frequency in terms of number of observations. This behaves
+        similarly to R's frequency for a time series (ts).
+
+    filter: np.array
+        A filter by which the convolution will be performed.
+
+    Returns
+    -------
+    decomposed_tuple : namedtuple
+        A named tuple with `x`, `trend`, `seasonal`, and `random` components
+        where `x` is the input signal, `trend` is the overall trend, `seasonal`
+        is the seasonal component, and `random` is the noisy component. The
+        input signal `x` can be mostly reconstructed by the other three
+        components with a number of points missing equal to `f`.
+
+    Notes
+    -----
+    This function is generally used in conjunction with
+    :func:`pmdarima.utils.visualization.decomposed_plot`,
+    which plots the decomposed components. Also there are two example notebooks
+    within the `examples` folder which mirror existing R examples.
+
+     References
+    ----------
+    .. [1] Example of decompose using both multiplicative and additive types:
+           https://anomaly.io/seasonal-trend-decomposition-in-r/index.html
+
+    .. [2] R documentation for decompose:
+           https://www.rdocumentation.org/packages/stats/versions/3.6.1/topics/decompose
+
+
+    """
+
+    multiplicative = "multiplicative"
+    additive = "additive"
+
+    # Helper function to stay consistent and concise based on 'type'
+    def _decomposer_helper(a, b, type):
```

Comment:
You shouldn't need to pass `type` (or `typ` if you rename it) since it's in a closure

Reply:
Done!